modules/ags/src/main/scala/lucuma/ags/AgsGuideQuality.scala

```diff
@@ -6,41 +6,45 @@ package lucuma.ags
 import cats.Order
 import lucuma.core.util.Enumerated
 
-sealed trait AgsGuideQuality extends Product with Serializable {
-  def message: String
-}
-
-object AgsGuideQuality {
-  case object DeliversRequestedIq   extends AgsGuideQuality {
-    override val message = "Delivers requested IQ."
-  }
-  case object PossibleIqDegradation extends AgsGuideQuality {
-    override val message = "Slower guiding required; may not deliver requested IQ."
-  }
-  case object IqDegradation         extends AgsGuideQuality {
-    override val message = "Slower guiding required; will not deliver requested IQ."
-  }
-  case object PossiblyUnusable      extends AgsGuideQuality {
-    override val message = "May not be able to guide."
-  }
-  case object Unusable              extends AgsGuideQuality {
-    override val message = "Unable to guide."
-  }
-
-  val All: List[AgsGuideQuality] =
+enum AgsGuideQuality(private val tag: String, val message: String):
+
+  case DeliversRequestedIq
+      extends AgsGuideQuality("delivers_requested_id", "Delivers requested IQ.")
+
+  case PossibleIqDegradation
+      extends AgsGuideQuality("possible_iq_degradation",
+                              "Slower guiding required; may not deliver requested IQ."
+      )
+
+  case IqDegradation
+      extends AgsGuideQuality("iq_degradation",
+                              "Slower guiding required; will not deliver requested IQ."
+      )
+
+  case PossiblyUnusable extends AgsGuideQuality("possible_unusable", "May not be able to guide.")
+
+  case Unusable extends AgsGuideQuality("unusable", "Unable to guide.")
+
+end AgsGuideQuality
+
+object AgsGuideQuality:
+
+  private val All: List[AgsGuideQuality] =
```

Comment:
For my own edification, is there a reason to keep this given the compiler-generated `values` method?

Reply:
Well, it establishes an order between the different quality levels. Maybe it should be called `ranking` rather than `All`